### PR TITLE
dyngen 0.2.0

### PR DESCRIPTION
**Pull request by rcannood**
* Complete rewrite of `dyngen`. Major improvements: 
 
   - All aspects of the pipeline have been optimised towards execution time and end-user usability.
   - `dyngen` 0.2.0 uses `fastgssa` 0.2.0, which has also been rewritten entirely in `Rcpp`,
     thereby improving the speed significantly.
   - Mapping of a simulation to the gold standard is greatly improved.
   - Custom backbones can be defined using backbone lego pieces. See `?bblego` for more information.

### CODE REVIEW

File: R/generate_dataset.R [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  model <- model %>% 
+    generate_tf_network() %>% 
+    generate_feature_network() %>% 
+    generate_kinetics() %>% 
+    generate_gold_standard() %>%  
+    generate_cells() %>% 
+    generate_experiment()
+  
+  dataset <-
+    wrap_dataset(model, store_grn = store_grn)
```
Comment (zouter) [inline R/generate_dataset.R]: Might it be worthwile to add some prior information here based on the gold standard?

File: README.Rmd [hunk 2]
```diff
@@ -11,46 +11,240 @@ knitr::opts_chunk$set(fig.path = "man/figures/README_", warning=FALSE, message=F
 
 [![Build Status](https://travis-ci.org/dynverse/dyngen.svg)](https://travis-ci.org/dynverse/dyngen) [![codecov](https://codecov.io/gh/dynverse/dyngen/branch/master/graph/badge.svg)](https://codecov.io/gh/dynverse/dyngen)
 
-A package to generating synthetic single-cell data starting from regulatory networks. The data is generated in several steps:
-![generation_overview](man/figures/README_generation_overview_v1.svg)
+A package for generating synthetic single-cell data starting from regulatory networks. The data is generated in several steps:
```
Comment (zouter) [inline README.Rmd]: Would be usefult o link to the exact kintetic model somewhere, e.g. the dynbenchmark sup note

**Comment by zouter** [comment]
A decrease in code coverage :fearful:

File: R/1_initialisation.R [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  distance_metric = "pearson",
+  tf_network_params = tf_network_default(),
+  feature_network_params = feature_network_default(),
+  kinetics_params = kinetics_default(),
+  gold_standard_params = gold_standard_default(),
+  simulation_params = simulation_default(),
+  experiment_params = experiment_snapshot(),
+  verbose = TRUE,
+  download_cache_dir = NULL,
+  num_cores = 1
```
[the remark this answers is not in the document]
Comment (rcannood) [reply R/1_initialisation.R]: true!

[the remark this answers is not in the document]
Comment (rcannood) [reply R/1_initialisation.R]: Can i turn this comment into an issue easily? ;P

File: R/generate_dataset.R (hunk 1 above)
Comment (rcannood) [reply R/generate_dataset.R]: true!

**Comment by rcannood** [comment]
Yeah. Although I would probably only need to test one function in order to be able to achieve better code coverage ;P

File: README.Rmd (hunk 2 above)
Comment (rcannood) [reply README.Rmd]: true, although this is already outdated by now, i think it should be included into the dyngen repo instead

File: R/1_initialisation.R (hunk 3 above)
[the remark this answers is not in the document]
Comment (zouter) [reply R/1_initialisation.R]: Yes I turned it into an issue by clicking the ... :slightly_smiling_face:

[the remark this answers is not in the document]
Comment (rcannood) [reply R/1_initialisation.R]: Ohhhhh! :bulb:

**Pull request merged**